widget/embedded/src/hooks/useLanguage.ts

```diff
@@ -9,18 +9,20 @@ interface UseLanguage {
   languages: LanguageItem[];
   defaultLanguage: Language;
   activeLanguage: Language;
-  changeLanguage: (language?: Language) => void;
+  changeLanguage: (language?: Language | null) => void;
+  resetLanguage: () => void;
 }
 
 export function useLanguage(): UseLanguage {
-  const { setLanguage, language } = useAppStore();
+  const { setLanguage, language, config } = useAppStore();
   const languages = LANGUAGES;
-  const defaultLanguage = DEFAULT_LANGUAGE;
+  const defaultLanguage = config?.language || DEFAULT_LANGUAGE;
 
   return {
-    activeLanguage: language,
+    activeLanguage: language || defaultLanguage,
```

Comment:
I believe that if the language in the configuration changes, the language of the widget should also change and take precedence over both the default and user-selected values. Currently, when a user selects a language in the widget, the widget configuration is unable to impact the language. I suggest aligning the behavior with the token selection behavior for a more desirable outcome.